include/fmt/format.h

```diff
@@ -51,7 +51,7 @@
 # define FMT_ICC_VERSION 0
 #endif
 
-#if (defined(__GNUC__) && !defined(__clang__) && __GNUC__ >= 406) || \
+#if (defined(__GNUC__) && !defined(__clang__) && (__GNUC__ * 100 + __GNUC_MINOR__) >= 406) || \
```

Comment:
I think you could use `FMT_GCC_VERSION` instead of `__GNUC__ * 100 + __GNUC_MINOR__` (possibly after reordering defines).

Reply:
`FMT_GCC_VERSION` is defined in core.h, which is included after this. This also needs to be before the inclusion of core.h to disable warnings from there. On your first review in #736 you said that `FMT_GCC_VERSION` needs to be in core.h.

So that leaves two options that I can think of:
1) Leave it as is
2) Duplicate these warning-disabling maneuvers in both core.h and format.h and move the inclusion of core.h before the disabling of the warnings in format.h, so both can use `FMT_GCC_VERSION`, like this:

```cpp
// core.h
#define FMT_GCC_VERSION /* ... */

#if FMT_GCC_VERSION
// disable
#endif

// ...

#if FMT_GCC_VERSION
// re-enable
#endif
```

```cpp
// format.h

#include "core.h"

#if FMT_GCC_VERSION
// disable
#endif

// ...

#if FMT_GCC_VERSION
// re-enable
#endif
```

Reply:
I said that very incoherently. What I was trying to say was that the warnings need to be disabled before the inclusion of core.h, but `FMT_GCC_VERSION` is defined in core.h. So we can either not use `FMT_GCC_VERSION` in disabling the warnings or disable them in both headers. The latter would probably be preferable, because now the warnings aren't disabled at all if you only include core.h.

Reply:
I think we can live without disabling these warnings in `fmt/core.h` for now, so I suggest just moving `fmt/core.h` to the front and using `FMT_GCC_VERSION` here.

Reply:
The original point of this PR was to fix the version check so there wouldn't be any -Wshadow warnings.

Reply:
Yes and this will fix the reported warnings in `fmt/format.h` (but not in `fmt/core.h`).